ament_cmake_gen_version_h/cmake/ament_cmake_gen_version_h.cmake

```diff
@@ -0,0 +1,136 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+# `ament_cmake_gen_version_h` function creates and installs a version header file.
+# The version is taken from `package.xml` file's `<version>` tag
+# The function uses a provided "version.h.in" template file to generate
+# the destination version file in the provided folder.
+# The generated file is being (re-)created if:
+# - the file does not exist
+# - the file does exists but contains a version that differs from the version in `package.xml` file
+#
+# :param NO_INSTALL: whether to autmatically install the generated version file into DESTINATION include
+# :type NO_INSTALL: BOOL
+# :default value NO_INSTALL: FALSE
+
+# :param INCLUDE_DIR: path to the include folder where the file will be generated
+#     ${INCLUDE_DIR} folder will be added to the include paths
+#     the file will be placed into ${INCLUDE_DIR}/${PROJECT_NAME} folder according to ROS2 standard
+# :type INCLUDE_DIR: string
+# :default value INCLUDE_DIR: ${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_gen_version_h/include
+
+# :param VERSION_FILE_NAME: file name of the generated header file
+# :type VERSION_FILE_NAME: string
+# :default value VERSION_FILE_NAME: version.h
+
+# :param VERSION_MAJOR: override VERSION_MAJOR
+# :type VERSION_MAJOR: string
+# :default value VERSION_MAJOR: VERSION_MAJOR from the package.xml file
+
+# :param VERSION_MINOR: file name of the generated header file
+# :type VERSION_MINOR: string
+# :default value VERSION_MINOR: VERSION_MINOR from the package.xml file
+
+# :param VERSION_PATCH: file name of the generated header file
+# :type VERSION_PATCH: string
+# :default value VERSION_PATCH: VERSION_PATCH from the package.xml file
+
+################################################################################
```

Comment:
@serge-nikulin nit: follow the docblock format that is used for other functions and macros in this repository.